src/core/ext/transport/chttp2/transport/chttp2_transport.c

```diff
@@ -518,7 +520,7 @@ static void init_transport(grpc_exec_ctx *exec_ctx, grpc_chttp2_transport *t,
                   &channel_args->args[i], settings_map[j].integer_options);
               if (value >= 0) {
                 push_setting(exec_ctx, t, settings_map[j].setting_id,
-                             (uint32_t)value);
+                             (uint32_t)value, true);
```

Comment:
It looks like we are now setting this parameter to false in all callers.  Given that, why not remove the parameter?  I think this entire PR could boil down to a 1-line change of simply removing the call to `grpc_chttp2_initiate_write()` from inside of `push_settings()`.